### ISSUE TITLE
[dv] Add ecc error generation to memory agent

### ISSUE THREAD

**Issue by GregAC** [lowRISC/ibex#1731]
We need to stimulate ECC errors. The incoming bus data has integrity bits. The driver should provide incorrect versions of these at configurable random internals in a similar manner to how bus errors work.

Also update co-simulation to support internal interrupts from ECC errors.

> estimate 8
> remaining 2022-09-20 3
> remaining 2022-10-14 1

**Comment by GregAC** [comment]
@ctopal decided to split this into two issues after all, see cosim issue here: https://github.com/lowRISC/ibex/issues/1732

I've left that unassigned for now. Potentially it's something I can look at as I'm more familiar with what's needed. Work on the agent side of things for now. I'll let you know if I find time for the cosim stuff and if not you can get going on that when you need it.

**Comment by ctopal** [comment]
>remaining 2022-09-20 3

**Comment by GregAC** [comment]
I have a set of commits on top of Canberk's work in https://github.com/lowRISC/ibex/pull/1811 but no PR yet. Once that is merged this will be complete.

**Comment by GregAC** [comment]
Completed by https://github.com/lowRISC/ibex/pull/1907
